### ISSUE TITLE
Improve the normalization of the features

### ISSUE THREAD

**Issue by LilySnow** [DeepRank/deeprank#7]
It is not clear how the normalization is done unless I read the code line by line. Is it done over mini-batch or batch or one case? Could we add more information to it? An equation?

**Comment by NicoRenaud** [comment]
The normalization is actually tricky to do as the entire dataset is never loaded in memory. So I have to compute the standard deviation, mean value and variance for individual molecule and then compute the std and mean value for the entire dataset. There is actually some info in the doc string of the NormParam class here:
https://github.com/DeepRank/deeprank/blob/master/deeprank/generate/NormalizeData.py 

This could be done in a better way and might be one of the improvement we need.

**Comment by LilySnow** [comment]
Thanks. We can also normalize the data over mini-batches. 

Another question about normalization. You mentioned in the document: "The normalization of the targets is done vian a min/max normalization." Why do you need to normalize dockQ, which is already between 0 and 1?

**Comment by NicoRenaud** [comment]
the one that are used are not necesarily between 0 and 1. You might have only very poor confs in your training set for example. Then again the normalization of the features/target is an option that can be turned off. Everything is in the doc now !

**Comment by sonjageorgievska** [comment]
We use batch normalization
